### PR TITLE
Feat/move frontend to api

### PR DESCRIPTION
**Pull request by AinoraZ**
- Implemented login page using API.
- Implemented topic list using API.
- Implemented User and All learning days using API.
- Made UserContext for ease of use when communicating to api.

### CODE REVIEW

File: schedulearn-frontend/src/UserContext.tsx [hunk 1]
```diff
@@ -57,6 +56,10 @@ class UserContextManager {
   }
 
   public async fetch(input: RequestInfo, init?: RequestInit): Promise<Response> {
```
Comment (AinoraZ) [inline schedulearn-frontend/src/UserContext.tsx:51]: Example usage: ```fetch("api/user/current")```;

**Review by DominykasP — APPROVED** [review]
👍

**Comment by AinoraZ** [comment]
[Login rewrite with API](https://trello.com/c/ugoCdIrJ/14-login-rewrite-with-api)

**Comment by AinoraZ** [comment]
[Rewrite TopicList to use API](https://trello.com/c/YVQzpcPA/16-rewrite-topiclist-to-use-api)

**Comment by AinoraZ** [comment]
[Rewrite LearningDayCalendar to use API](https://trello.com/c/1RfNmz8k/17-rewrite-learningdaycalendar-to-use-api)

**Pull request merged**